### PR TITLE
Add header_converters: support

### PR DESCRIPTION
**Pull request by olleolleolle**
This is inspired by CSV.parse in Ruby's standard library.

### CODE REVIEW

File: lib/esv.rb [hunk 1]
```diff
@@ -16,7 +16,27 @@ def self.generate_file(path, &block)
...
+    downcase: ->(value) { value.downcase },
+    # Details:
+    #
+    #  Strips leading and trailing whitespace.
+    #  Downcases the header.
+    #  Replaces embedded spaces with underscores.
+    #  Removes non-word characters.
+    #  Makes the string into a Symbol.
+    symbol: ->(value) {
+      value.strip.downcase.tr(" ", "_").gsub(/W+/, "").to_sym
```
Comment (henrik) [inline lib/esv.rb:32]: `W` should be `\W`. Missing spec coverage?

File: spec/esv_spec.rb [hunk 2]
```diff
@@ -72,6 +72,57 @@
...
+        ESV::HEADER_CONVERTERS[:upcase] = ->(value) { value.upcase }
+
+        output = ESV.parse(data, header_converters: :upcase)
+
+        expect(output).to eq [
+          [ "DOGS", "CATS CATS" ],
+          [ 1, 2 ],
+        ]
+
+        ESV::HEADER_CONVERTERS.delete(:upcase)
```
Comment (henrik) [inline spec/esv_spec.rb:111]: Minor: I wonder if putting this in an `ensure` would work to ensure it runs even if that spec fails. Or would we need to use `after`?

File: lib/esv.rb (hunk 1 above)
Comment (olleolleolle) [reply lib/esv.rb:32]: You're absolutely right! @olleolleolle, add a test or two for the converters, perhaps just as individual unit tests of those procs.

File: spec/esv_spec.rb (hunk 2 above)
Comment (olleolleolle) [reply spec/esv_spec.rb:111]: I'd say it's easier to use ans read an after block for this.

@olleolleolle Add more contexts!

**Pull request merged**